noria-server/src/controller/inner.rs

```diff
@@ -732,11 +732,18 @@ impl ControllerInner {
             Err(_) => {
                 // if the recipe doesn't know about this query, traverse the graph.
                 // we need this do deal with manually constructed graphs (e.g., in tests).
-                *self.outputs().get(name)?
+                if let Some(ni) = self.outputs().get(name) {
+                    *ni
+                } else {
+                    // depending on how the graph was constructed, the outputs may be suffixed
+                    // with the reader index.
+                    let reader_name = format!("{}_0", name);
+                    *self.outputs().get(&reader_name)?
```

Comment:
Doesn't this mean that tests will only ever use the first replica? Why is it correct to just always use `_0` here?

Reply:
`outputs()` returns a map from reader name to "the the node address that is being materialized not that of the reader node itself". So for something like `it_works_with_simple_arithmetic()` all of `CarPrice_0`, `CarPrice_1`, `CarPrice_2` map to the same node. From there we choose a replica from the list of replicas cached in the node. In a test like `add_columns()` I guess the graph is constructed manually(?) and the readers have a name like `a` instead of `a_0`.

Reply:
Ohhhhhh, I see what you mean. You're not actually looking for the `Reader` here, but their ancestor. Makes sense.